src/ssl.c

```diff
@@ -15934,13 +16029,39 @@ WOLFSSL_API int X509_PUBKEY_get0_param(WOLFSSL_ASN1_OBJECT **ppkalg, const unsig
     return SSL_FAILURE;
 }
 
-/*** TBD ***/
+#if defined(OPENSSL_EXTRA)
+#if !defined(NO_FILESYSTEM) && !defined(NO_CERTS) && !defined(NO_RSA)
 WOLFSSL_API WOLFSSL_EVP_PKEY *wolfSSL_get_privatekey(const WOLFSSL *ssl)
 {
-    (void)ssl;
-    WOLFSSL_STUB("SSL_get_privatekey");
-    return NULL;
+
+  if (ssl == NULL) {
+      return NULL;
+  }
+
+  if (ssl->buffers.weOwnKey) {
+          if (ssl->buffers.key == NULL) {
```

Comment:
Indentation spacing is off in this function, please fix.  Our coding standard states that we should use spaces only (no hard tabs), with each tab being 4 spaces.